packages/did/src/DidLinks/AccountLinks.chain.ts

```diff
@@ -0,0 +1,243 @@
+/**
+ * Copyright 2018-2021 BOTLabs GmbH.
+ *
+ * This source code is licensed under the BSD 4-Clause "Original" license
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+import { BlockchainApiConnection } from '@kiltprotocol/chain-helpers'
+import type {
+  Deposit,
+  IDidIdentifier,
+  IIdentity,
+  KeyringPair,
+  KeystoreSigner,
+  SubmittableExtrinsic,
+  VerificationKeyType,
+} from '@kiltprotocol/types'
+
+import type { SignerOptions } from '@polkadot/api/types'
+import type { Null, Option, Struct } from '@polkadot/types'
+import type { AccountId, Extrinsic } from '@polkadot/types/interfaces'
+import type { AnyNumber } from '@polkadot/types/types'
+import type { HexString } from '@polkadot/util/types'
+
+import type { DidAuthorizationOptions, FullDidDetails } from '../DidDetails'
+import { getKiltDidFromIdentifier, parseDidUri } from '../Did.utils.js'
+
+// TODO: update with string pattern types once available
+type AccountAddress = IIdentity['address']
+type DidUri = FullDidDetails['did']
+
+interface ConnectionRecord extends Struct {
+  did: AccountId
+  deposit: Deposit
+}
+
+interface DepositJSON {
+  amount: number
+  owner: AccountAddress | null
+}
+
+/* ### QUERY ### */
+
+export async function getAccountLinkDepositInfo(
+  linkedAccount: AccountId | string
+): Promise<DepositJSON> {
+  const blockchain = await BlockchainApiConnection.getConnectionOrConnect()
+  const connectedDid = await blockchain.api.query.didLookup.connectedDids<
+    Option<ConnectionRecord>
+  >(linkedAccount)
+  const { owner, amount } = connectedDid.unwrapOrDefault().deposit
+  return {
+    amount: amount.toNumber(),
+    owner: connectedDid.isSome ? owner.toString() : null,
+  }
+}
+
+export async function getConnectedDidForAccount(
+  linkedAccount: AccountId | string
+): Promise<DidUri | null> {
```

Comment:
This is a chain file, so I am kind of against having types such as `DidUri`, when it should be just a `DidIdentifier`. If we want to use `DidUri`, we should at least also add functions that simply take or return the `DidIdentifier`.

Reply:
The point is, in my opinion, that this is a chain file, and as a chain file it should not know how to create a full DID given the identifier. That is unnecessary coupling. So a good compromise would be provide both the raw identifier and the compounded DID.

Reply:
ok, no more didUris